docs/guides/implement_custom_components.md

```diff
@@ -0,0 +1,171 @@
+# Guide - Implement custom components
+
+**Level**: Beginner
+**Time**: 20min
```

Comment:
This was just a rough estimation from my end. Wdyt, is this realistic?

Reply:
maybe we can just count the average reading time of the text and add 5-10min ?

Reply:
pfffrt I can do it in 5 easy